src/main/scala/com/ebiznext/comet/job/index/bqload/BigQuerySparkJob.scala

```diff
@@ -31,14 +31,33 @@ class BigQuerySparkJob(
     extends SparkJob
     with BigQueryJobBase {
 
-  override def withExtraSparkConf(): Map[String, String] =
-    if (cliConfig.writeDisposition == "WRITE_TRUNCATE")
-      cliConfig.options ++ Map(
-        "spark.datasource.bigquery.allowFieldAddition"   -> "false",
-        "spark.datasource.bigquery.allowFieldRelaxation" -> "false"
-      )
-    else
-      cliConfig.options
+  val bqOptions = {
+    val schemaUpdateOptions = List(
+      "spark.datasource.bigquery.allowFieldAddition",
+      "spark.datasource.bigquery.allowFieldRelaxation",
+      "allowFieldAddition",
+      "allowFieldRelaxation"
+    )
+    Option(bigquery.getTable(tableId)) match {
+      case None =>
+        cliConfig.options -- schemaUpdateOptions
+      case Some(tableId) if cliConfig.writeDisposition == "WRITE_TRUNCATE" =>
+        val definition = tableId.getDefinition.asInstanceOf[StandardTableDefinition]
+        if (definition.getTimePartitioning == null && definition.getRangePartitioning == null)
+          cliConfig.options -- schemaUpdateOptions
+        else
+          (cliConfig.options -- schemaUpdateOptions) ++ Map(
+            "spark.datasource.bigquery.allowFieldAddition"   -> "false",
+            "spark.datasource.bigquery.allowFieldRelaxation" -> "false"
+          )
+      case Some(_) =>
+        cliConfig.options
+    }
+  }
+
+  override def withExtraSparkConf(): Map[String, String] = {
+    bqOptions
```

Comment:
I am wondering if we should not simply assume that the user does not use the spark property by default and when he uses it it means that he wants to force the alloField* semantic.

Reply:
Following our conversation
I've differentiated between the connector options and the spark options
I simplified the way we add options to the reader/writer using the options (you were right it is available for some time - since 1.4)

Reply:
I didn't get to read all the modifications yet but I think that this PR and previous commits that introduced the method `withExtraSparkConf()`  are trying to handle an issue that should be handled by the user directly. Using allowField* options leads to a BQ error if we are in an Write_Truncate Mode but this is the expected behavior.  I think that KISS principle can fairly be applied here, a wrong configuration options leads to an error and it is up to user to use the right options (whether using spark session or bigquery connector) . Adding code to change the explicit configuration passed by the user adds complexity and can be a source of unexplained behavior (from user point of view) . What do you think ?

Reply:
@mhdkassir We are trying to solve the following issue:
Assume you have a non partitioned table and still is willing to allow field addition/relaxation. This would fail with BigQuery if we do nothing.

This is one of the case we are trying to handle.

Reply:
@hayssams perhaps I'm missing something, but if the table is not partitioned and the write mode is write_truncate field addition is possible without adding any options since bigquery will overwrite the schema

Reply:
@mhdkassir When you add data to the table for the first time, you get an error wether you set the allow* option fro true or false.
Below is the message you get from BQ
```Caused by: com.google.cloud.spark.bigquery.repackaged.com.google.cloud.bigquery.BigQueryException: Schema update options should only be specified with WRITE_APPEND disposition, or with WRITE_TRUNCATE disposition on a table partition.```
In that case we need to remove completely the option.

Reply:
@hayssams yes but should it be removed by the application code ? I think it is the responsibility of the user to simply correct the configuration by not setting allowField* options at all in such case. I think starlake should not undo an explicit user configuration to mask third party errors. Up to the user to use BigQuery correctly.

Reply:
But then the user need to set different values on first ingestion and for the next ones

Reply:
@mhdkassir When we have a merge operation, the user request an append mode and since he wants field addition/relaxation, he will set those options trop true.
But in fact we end up merge the datasets in memory and do a write_truncate. IF we do no reset those allow* fields, we get the errors above.

Reply:
@seyguai found an interesting bug that would have made us loose the optimisation released with 0.2.6
https://github.com/starlake-ai/starlake/pull/32/files#diff-02d5a7d6430888c78f970e0494a55e832db4a790b8ec2b66c4168a1f012bccfcL908